readthedocs/projects/version_handling.py

```diff
@@ -158,3 +161,86 @@ def determine_stable_version(version_list):
         version_obj, comparable = versions[0]
         return version_obj
     return None
+
+
+def sort_versions_python_packaging(version_list):
+    """
+    Sort Read the Docs versions list using ``packaging`` algorithm.
+
+    All the invalid version (raise ``InvalidVersion``) are added at the end
+    sorted alphabetically.
+
+    https://pypi.org/project/packaging/
+    https://packaging.python.org/en/latest/specifications/version-specifiers/
+    """
+    alphabetically_sorted_version_list = sorted(
+        version_list,
+        key=operator.attrgetter("slug"),
```

Comment:
We should probably use `verbose_name` for sorting.

Reply:
Hrm... Good point! I would keep this change for another PR because I've been using slug in multiple places on the addons API (example: https://github.com/readthedocs/readthedocs.org/blob/96fc90bfdc318c962e14c6e92da16bd55cd176b5/readthedocs/proxito/views/hosting.py#L333-L341) and we are using `slug` to compare things in the addons JavaScript as well: https://github.com/readthedocs/readthedocs-client/blob/7a1735049289d1c8c7d706f741aa092e1190d5ec/src/flyout.js#L199-L206

So, we would need to change all of these places and it will require a bigger change here.

Reply:
I created an issue for this at https://github.com/readthedocs/readthedocs.org/issues/11182